### ISSUE TITLE
DialogContextError: DialogStateManager.getValue: a scope of 'json' wasn't found

### ISSUE THREAD

**Issue by zachgoz** [microsoft/BotFramework-Composer#7834]
## Describe the bug

After installing the custom runtime I am seeing this errors after making a HTTP request. The initial error is:

value:"DialogContextError: DialogStateManager.getValue: a scope of 'json' wasn't found."
if I change the response type to blank then I see: 
DialogContextError: args[0].trim is not a function

## Version

Bot Framework Composer

Version:  1.4.1
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS
- [ x] macOS
- [ ] Windows
- [ ] Ubuntu

## Expected behavior

The HTTP request is successful and the response is saved to a variable

## Screenshots

<img width="1266" alt="Screen Shot 2021-05-14 at 12 50 52 PM" src="https://user-images.githubusercontent.com/3943365/118303293-47b32300-b4b3-11eb-815e-6367f9d93e5f.png">
<img width="851" alt="Screen Shot 2021-05-14 at 12 53 21 PM" src="https://user-images.githubusercontent.com/3943365/118303384-61ed0100-b4b3-11eb-9064-62251e9607cb.png">

## Additional context

This was working fine before adding the custom runtime.

**Comment by zachgoz** [comment]
It appears this is only an issue with the Node JS runtime. When build composer from the source and run the latest version I have an option to convert to a composer 2.0 project. If I do this and use the NodeJS runtime I see these errors, if I use the C# runtime the dialog works as expected.

**Comment by ninggao** [comment]
@anusharavindrar Could you please take a look at this one?

**Comment by zachgoz** [comment]
Is there any other info I could provide to help solve this?

**Comment by jwiley84** [comment]
I am unable to repro this using the electron version. I will try  the source version and see if it repros there.

**Comment by zachgoz** [comment]
This seems to be an issue for any nodejs runtime, I've tried node 12.20 and 14.17 both fresh installs of composer and existing and new composer projects have the same errors. Is the node runtime not supported anymore?

<img width="1170" alt="Screen Shot 2021-05-20 at 2 01 46 PM" src="https://user-images.githubusercontent.com/3943365/119035710-e04b1680-b97d-11eb-94b3-3a8bccd413d3.png">

**Comment by ninggao** [comment]
@jwiley84 Any updates on this one?

**Comment by jwiley84** [comment]
I am unable to repro this on the source version. I'm going to ask a member of the composer team to chime in.

**Comment by zachgoz** [comment]
Ok I made some progress here. So the first error "a scope of json wasn't found" does not show up when I do not set json as the response type in the nodejs runtime. In C this does not seem to matter.
<img width="376" alt="Screen Shot 2021-05-27 at 3 58 45 PM" src="https://user-images.githubusercontent.com/3943365/119889383-7ba33500-bf04-11eb-83fe-7041564f93bf.png">

This uncovered a previous issue where certain values that should have been objects were being returned as strings so I had to use the =json() function to convert them to objects I could access. It appears this is fixed in the latest release and leaving the =json() around a non string throws the args[0].trim error because the input is not a string.
<img width="1075" alt="Screen Shot 2021-05-27 at 4 21 19 PM" src="https://user-images.githubusercontent.com/3943365/119892019-a17e0900-bf07-11eb-960c-c24379661cae.png">

It would be helpful if the =json() function would catch an error if a non-string is passed to it. Or maybe it would be better to just process it instead:

(typeof args[0] === 'string') ? JSON.parse(args[0].trim()) : args[0]
<img width="1012" alt="Screen Shot 2021-05-27 at 4 24 59 PM" src="https://user-images.githubusercontent.com/3943365/119892764-88298c80-bf08-11eb-88b7-888fdb8c9255.png">

**Comment by jamalsia** [comment]
i have the same issue.

**Comment by johnataylor** [comment]
@joshgummersall can you take a look? thanks

**Comment by joshgummersall** [comment]
Yep - I can repro, so something wonky is going on.

**Comment by joshgummersall** [comment]
@zachgoz, we are still working on a resolution for this. Did the temporary fix above solve the problem for now?

**Comment by zachgoz** [comment]
Yes, thank you for the suggestion and update. This workaround got it working for me.

**Comment by joshgummersall** [comment]
We are fixing this at the SDK level as well, so when that updates it'll be fully resolved. Thanks for reporting the issue!
